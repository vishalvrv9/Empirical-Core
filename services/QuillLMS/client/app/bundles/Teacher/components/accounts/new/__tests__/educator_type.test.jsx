import React from 'react';
import { shallow, mount } from 'enzyme';

import EducatorType from '../educator_type';

import SchoolSelector from '../../../shared/school_selector'
import NotUsK12View from '../../school/not_us_k12_view'

describe('EducatorType component', () => {

  it('should render if stage is 1', () => {
      const wrapper = shallow(
        <EducatorType />
      );
      expect(wrapper).toMatchSnapshot();
  });

  it('should render SchoolSelector if stage state is 2', () => {
      const wrapper = shallow(
        <EducatorType />
      );
      wrapper.setState({ stage: 2 });
      expect(wrapper.find(SchoolSelector)).toHaveLength(1);
  });

  it('should render NotUsK12View if stage state is 3', () => {
      const wrapper = shallow(
        <EducatorType />
      );
      wrapper.setState({ stage: 3 });
      expect(wrapper.find(NotUsK12View)).toHaveLength(1);
  });

  it('should be teachers/classrooms/google_sync if teacherFromGoogleSignUp prop is true', () => {
    // Object.defineProperty(window.location, 'pathname', {
    //   writable: true,
    //   value: '/'
    // });
    // const wrapper = mount(
    //   <EducatorType
    //     teacherFromGoogleSignUp={Boolean(true)}
    //     analytics = {{ track: function(){} }} />
    // )
    // const k12View = wrapper.find(UsK12View)
    // k12View.setState({selectedSchool: {name: 'df'}})
    // expect(wrapper.find(UsK12View).find('.select_school_button').length).toBe(1);

    // expect(window.location.pathname).toBe('/teachers/classrooms/google_sync')
  })

});
