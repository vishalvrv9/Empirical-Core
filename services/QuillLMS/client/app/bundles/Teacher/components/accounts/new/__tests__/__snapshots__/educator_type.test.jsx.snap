// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducatorType component should render if stage is 1 1`] = `
<div
  className="educator-type"
>
  <ScrollToTop />
  <h3>
    Are you a faculty member at a U.S. K-12 school?*
  </h3>
  <div
    className="option-wrapper"
  >
    <button
      className="button contained primary large"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="button contained primary large"
      onClick={[Function]}
    >
      No
    </button>
  </div>
  <div
    className="explanation"
  >
    *K-12 is a term for school grades prior to college.
    <br />
    These grades span from kindergarten through the 12th grade.
  </div>
</div>
`;
